src/output-json.c

```diff
@@ -981,13 +981,23 @@ int OutputJsonBuilderBuffer(
     size_t jslen = jb_len(js);
     DEBUG_VALIDATE_BUG_ON(jb_len(js) > UINT32_MAX);
     if (MEMBUFFER_OFFSET(*buffer) + jslen >= MEMBUFFER_SIZE(*buffer)) {
-        MemBufferExpand(buffer, (uint32_t)jslen);
+        size_t expand_by = MEMBUFFER_OFFSET(*buffer) + jslen + 1;
```

Comment:
git message says "This also fixes the call to MemBufferExpand which is supposed be
passed the amount to expand by, not the new size required." , however it seems the opposite happens here: total is passed.

Wonder if we should just grow to a next multiple of 4k (or PAGESIZE) that fits